From: SeongJae Park <sj@kernel.org>
Date: Fri, 14 Feb 2025 17:23:32 -0800
Subject: [PATCH] ==== make allow filters after reject filters useful ====

mm/damon: let allow filters after reject filters useful

DAMOS filters do allow or reject elements of memory for given DAMOS
action only if those match the condition of filters.  For elements that
don't match any DAMOS filter, 'allowing' is the default behavior.
Hence, allow DAMOS filters that don't have any reject DAMOS filter after
them are just meaningless overhead.  The decision was made to keep the
behavior consistent with a case that no filter is installed at all.
This makes usage of DAMOS filters confusing and inefficient, though.  If
there is no filter at all or the last filter is a reject filter,
allowing by default makes sense.  But if the last filter is an allow
filter, rather rejecting by default is apparently more intuitive
behavior in a common sense.

Update the filtering logic to work in the way.  Decide the default
behavior as the opposite of the last isntalled filter.  If it is a
reject filter, allow by default, and vice versa.

Note that this is changing the old behavior.  But the old behavior was
definitely confusing, inefficient and anyway useless.  Also, the
behavior has relatively recently introduced.  It is difficult to
anticipate any user that depends on the behavior.  Hence this is not a
user-breaking behavior change but obvious improvement.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_f274e6so | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_f274e6so

diff --git a/hkml_cv_bogus/hkml_cv_bogus_f274e6so b/hkml_cv_bogus/hkml_cv_bogus_f274e6so
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.39.5

